### PR TITLE
[support presigned URL] support http request without HTTP_USER_AGENT_FIELD

### PR DESCRIPTION
**Pull request by giuspen**
Support http request without HTTP_USER_AGENT_FIELD

Description
-----------
As discussed in https://github.com/aws/aws-iot-device-sdk-embedded-C/issues/1887 I have made changes to an existing demo http_demo_s3_download to generate a presigned URL instead of directly downloading the file.
I'm planning, if the changes to the submodules are accepted, to then generate a new demo called http_demo_s3_presigned_url or something like that.
You can see/test my work at https://github.com/giuspen/aws-iot-device-sdk-embedded-C/tree/GP_http_demo_s3_download_test_signature (only generate presigned URL) or https://github.com/giuspen/aws-iot-device-sdk-embedded-C/tree/GP_http_demo_s3_download_test_signature2 (download and also generate presigned URL)

Test Steps
-----------
Exact same requirements of original demo http_demo_s3_download

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------
https://github.com/aws/aws-iot-device-sdk-embedded-C/issues/1887
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by moninom1** [comment]
Hello @giuspen, Thank you for raising the PR! We will be looking further into this issue as soon as we can, thank you for your patience!”

File: source/include/core_http_client.h [hunk 1]
```diff
@@ -111,7 +111,19 @@
...
  * This flag is valid only for #HTTPRequestInfo_t reqFlags parameter.
  */
-#define HTTP_REQUEST_KEEP_ALIVE_FLAG    0x1U
+#define HTTP_REQUEST_KEEP_ALIVE_FLAG       0x1U
+
+/**
+ * @ingroup http_request_flags
+ * @brief Set this flag to indicate that the request is for a persistent
+ * connection.
+ *
```
Comment (archigup) [inline source/include/core_http_client.h:120]: This comment seems to need to be updated

**Review by archigup — APPROVED** [review]

**Review by archigup — APPROVED** [review]

**Review by AniruddhaKanhere — APPROVED** [review]
Thank you for these changes and the tests @giuspen.

The changes look good to me.

**Pull request merged**